### PR TITLE
Bug 1112131 - Start end attributes rtl friendly

### PR DESCRIPTION
**Pull request by julienw**

### CODE REVIEW

File: dist/gaia-header.js [hunk 1]
```diff
@@ -677,6 +668,22 @@ return w[n];},m.exports,m);w[n]=m.exports;};})('gaia-header',this));
 },{"./lib/font-fit":4,"gaia-component":1,"gaia-icons":2}],4:[function(require,module,exports){
 ;(function(define){'use strict';define(function(require,exports,module){
 
+  var privMap = new WeakMap();
```
Comment (wilsonpage) [inline dist/gaia-header.js]: Can you comment to explain what this is for?

File: dist/gaia-header.js [hunk 2]
```diff
@@ -169,42 +86,31 @@ function firstChildTextNode(el) {
...
- * Some CSS rules, such as @keyframes
- * and @font-face don't work inside
- * scoped or shadow <style>. So we
- * have to put them into 'global'
- * <style> in the head of the
- * document.
- *
- * @param  {String} css
- */
-function injectGlobalCss(css) {
```
Comment (wilsonpage) [inline dist/gaia-header.js]: Why has this function been removed?

File: dist/gaia-header.js [hunk 3]
```diff
@@ -2,9 +2,6 @@
 ;(function(define){define(function(require,exports,module){
```
Comment (wilsonpage) [inline dist/gaia-header.js]: All changes to [gaia-component](http://github.com/gaia-components/gaia-component) must be landed there.

File: dist/gaia-header.js [hunk 4]
```diff
@@ -14,39 +11,25 @@ var noop  = function() {};
  * @return {Boolean}
  */
 var hasShadowCSS = (function() {
-  var div = document.createElement('div');
-  try { div.querySelector(':host'); return true; }
+  try { document.querySelector(':host'); return true; }
```
Comment (wilsonpage) [inline dist/gaia-header.js]: I originally changed this on your request.

Comment (wilsonpage) [inline dist/gaia-header.js]: Looks like perhaps you built without updating your local version of `gaia-component`?

Comment (julienw) [inline dist/gaia-header.js]: yeah, don't pay attention to these  changes :)

File: dist/gaia-header.js (hunk 1 above)
Comment (julienw) [inline dist/gaia-header.js]: Yep right, I can add a comment.

The goal is to hold values for a specific heading.

File: lib/font-fit.js [hunk 5]
```diff
@@ -1,5 +1,25 @@
 ;(function(define){'use strict';define(function(require,exports,module){
 
+  /* This map holds values specific to a heading.
+   * The key is the heading node itself. Because it's a WeakMap, it should be
+   * freed by the GC when the node is removed from the DOM.
+   */
+  var privMap = new WeakMap();
+
+  function getPriv(instance) {
```
Comment (wilsonpage) [inline lib/font-fit.js]: This seems like quite a complicated way to cache `start` and `end` attributes? Can't we just store them on the heading instance?

```js
heading.dataset.start = start;
heading.dataset.end = end;
```

File: gaia-header.js [hunk 6]
```diff
@@ -65,35 +94,91 @@ module.exports = Component.register('gaia-header', {
...
+  /**
+   * This function will debounce the use of runFontFit. This is used in
+   * attributeChanged so that the component user can change different attributes
+   * and still have runFontFit run once.
+   *
+   * @private
+   */
+  runFontFitSoon: function() {
+    clearTimeout (this.runFontFitTimeout);
+    this.runFontFitTimeout = setTimeout(() => this.runFontFit());
```
Comment (wilsonpage) [inline gaia-header.js]: Is there a risk when attached, we could see a paint before this gets run?

Comment (julienw) [inline gaia-header.js]: I don't use runFontFitSoon in attached, I use it only in attributeChanged

File: dist/gaia-header.js [hunk 7]
```diff
@@ -701,22 +809,65 @@ return w[n];},m.exports,m);w[n]=m.exports;};})('gaia-header',this));
...
+      if (start !== undefined) {
+        priv.start = start;
+      }
+
+      if (end !== undefined) {
+        priv.end = end;
+      }
+
+      if (width !== undefined) {
+        priv.width = width;
```
Comment (wilsonpage) [inline dist/gaia-header.js]: `width` is `undefined`

Comment (julienw) [inline dist/gaia-header.js]: :D was very rough draft ;)

File: gaia-header.js [hunk 8]
```diff
@@ -18,6 +18,8 @@ require('gaia-icons');
  */
 var actionTypes = { menu: 1, back: 1, close: 1 };
 
+const KNOWN_ATTRIBUTES = ['action', 'available-width', 'no-font-fit', 'title-start', 'title-end'];
```
Comment (gmarty) [inline gaia-header.js]: So, the `available-width` is not used or am I missing something?

**Pull request closed without merging**